core/src/main/java/org/geysermc/geyser/level/physics/BoundingBox.java

```diff
@@ -142,6 +142,47 @@ private boolean checkOverlapInAxis(double xOffset, double yOffset, double zOffse
         };
     }
 
+    public void pushOutOfBoundingBox(BoundingBox playerBox, Direction direction, double maxPushTolerance) {
```

Comment:
Is there a specific reason we're not pushing the player at all if it's at or higher as opposed to pushing the player by the max push distance distance?